src/transformers/models/qwen2/modeling_qwen2.py

```diff
@@ -332,7 +332,7 @@ def forward(self, x, position_ids):
         device_type = x.device.type
         device_type = device_type if isinstance(device_type, str) and device_type != "mps" else "cpu"
         with torch.autocast(device_type=device_type, enabled=False):
-            freqs = (inv_freq_expanded.float().to(x.device) @ position_ids_expanded.float()).transpose(1, 2)
+            freqs = (inv_freq_expanded.float().to(x.device) @ position_ids_expanded.float().to(x.device)).transpose(1, 2)
```

Comment:
We can improve performance slightly by doing these together:
```suggestion
            freqs = (inv_freq_expanded.float().to(x.device) @ position_ids_expanded.to(x.device, torch.float32)).transpose(1, 2)
```